src/line.rs

```diff
@@ -1062,6 +1123,72 @@ impl<'input, Endian> LineNumberProgramHeader<'input, Endian>
         StateMachine::new(self)
     }
 
+    /// Execute the line number program, completing the LineNumberProgramHeader
+    /// and producing an array of sequences within the line number program that
+    /// can later be used with `resume_from`.
+    ///
+    /// ```
+    /// # fn foo() {
+    /// use gimli::{LineNumberProgramHeader, NativeEndian};
+    ///
+    /// fn get_line_number_program_header<'a>() -> LineNumberProgramHeader<'a, NativeEndian> {
+    ///     // Get a line number program header from some offset in a
+    ///     // `.debug_line` section...
+    /// #   unimplemented!()
+    /// }
+    ///
+    /// let header = get_line_number_program_header();
+    /// let (complete_header, sequences) = header.sequences().unwrap();
+    /// println!("There are {} sequences in this line number program", sequences.len());
+    /// # }
+    /// ```
+    pub fn sequences(self)
+                     -> parser::Result<(LineNumberProgramHeader<'input, Endian>,
+                                        Vec<LineNumberSequence<'input, Endian>>)> {
```

Comment:
Do we want to let the caller control allocation (maybe they know when they can reuse a vec?) by accepting an `&mut Vec<LineNumberSequence<'input, Endian>>` parameter, asserting it is initially empty, and then filling it?

I'm not actually 100% sure if this would actually work or not due to the `'input` lifetime.